baselines/heterofl/heterofl/models.py

```diff
@@ -0,0 +1,705 @@
+            hidden_size[1],
+            num_blocks[1],
+            stride=2,
+            rate=rate,
+            norm=norm,
+            track=track,
+        )
+        self.layer3 = self._make_layer(
+            block,
+            hidden_size[2],
+            num_blocks[2],
+            stride=2,
+            rate=rate,
+            norm=norm,
+            track=track,
+        )
+        self.layer4 = self._make_layer(
+            block,
+            hidden_size[3],
+            num_blocks[3],
+            stride=2,
+            rate=rate,
+            norm=norm,
+            track=track,
+        )
+
+        self.classes_size = num_classes
+        self.mask = mask
+
+        if norm == "bn":
+            n_4 = nn.BatchNorm2d(
+                hidden_size[3] * block.expansion,
+                momentum=None,
+                track_running_stats=track,
+            )
+        elif norm == "in":
+            n_4 = nn.GroupNorm(
+                hidden_size[3] * block.expansion, hidden_size[3] * block.expansion
+            )
+        elif norm == "ln":
+            n_4 = nn.GroupNorm(1, hidden_size[3] * block.expansion)
+        elif norm == "gn":
+            n_4 = nn.GroupNorm(4, hidden_size[3] * block.expansion)
+        elif norm == "none":
+            n_4 = nn.Identity()
+        else:
+            raise ValueError("Not valid norm")
+        self.n_4 = n_4
+        if scale:
+            self.scaler = _Scaler(rate)
+        else:
+            self.scaler = nn.Identity()
+        self.linear = nn.Linear(hidden_size[3] * block.expansion, num_classes)
+
+    def _make_layer(self, block, planes, num_blocks, stride, rate, norm, track):
+        strides = [stride] + [1] * (num_blocks - 1)
+        layers = []
+        for strd in strides:
+            layers.append(block(self.in_planes, planes, strd, rate, norm, track))
+            self.in_planes = planes * block.expansion
+        return nn.Sequential(*layers)
+
+    def forward(self, input_dict):
+        """Forward pass of the ResNet.
+
+        Parameters
+        ----------
+        input_dict : Dict
+            Dict that contains Input Tensor that will pass through the network.
+            label of that input to calculate loss.
+            label_split if masking is required.
+
+        Returns
+        -------
+        Dict
+            The resulting Tensor after it has passed through the network and the loss.
+        """
+        output = {}
+        x = input_dict["img"]
+        out = self.conv1(x)
+        out = self.layer1(out)
+        out = self.layer2(out)
+        out = self.layer3(out)
+        out = self.layer4(out)
+        out = F.relu(self.n_4(self.scaler(out)))
+        out = F.adaptive_avg_pool2d(out, 1)
+        out = out.view(out.size(0), -1)
+        out = self.linear(out)
+        if "label_split" in input_dict and self.mask:
+            label_mask = torch.zeros(self.classes_size, device=out.device)
+            label_mask[input_dict["label_split"]] = 1
```

Comment:
Hi @jafermarq ,

Here the `input_dict['label_split']` is a tensor of integer (_which was previously float_) values (_say [2 ,5]_), so the `label_mask[input_dict['label_split']]` would assign ones to `label_mask[2]` , `label_mask[5]`.

The error encountered is because I missed to convert label_split from float tensor to integer tensor in the last merge.
Please inform me if there are any other aspects I might have missed, and let me know if you are now able to execute the code without any issues.